source/Background/Controller.js

```diff
@@ -143,5 +144,65 @@ backgroundController.exposeController(
     callback(null, requests, [{ portId, callId }]);
   },
 );
+*/
+
+backgroundController.exposeController(
+  'requestBalance',
+  async (opts, metadata, accountId) => {
+    const { callback, message, sender } = opts;
+    const { id: callId } = message.data.data;
+    const { id: portId } = sender;
+
+    const keyringHandler = getKeyringHandler(
+      HANDLER_TYPES.GET_BALANCE,
+      keyring,
+    );
+    const icpBalance = await keyringHandler(accountId);
+
+    callback(null, icpBalance, [{ portId, callId }]);
+  },
+);
+
+backgroundController.exposeController(
+  'requestTransfer',
+  async (opts, metadata, args) => {
+    const { message, sender } = opts;
+    const { id: callId } = message.data.data;
+    const { id: portId } = sender;
+
+    const url = qs.stringifyUrl({
+      url: 'transfer.html',
+      query: {
+        callId,
+        portId,
+        metadataJson: JSON.stringify(metadata),
+        argsJson: JSON.stringify(args),
+      },
+    });
+
+    extension.windows.create({
+      url,
+      type: 'popup',
+      width: TRANSFER_SIZES.width,
+      height: TRANSFER_SIZES.detailHeightSmall,
+      top: 65,
+      left: metadata.pageWidth - TRANSFER_SIZES.width,
+    });
+  },
```

Comment:
we are missing the connected check here

Reply:
`requestBalance` checks connection through keyringHandler
`requestTransfer` added in tbdf2a2b